src/main/java/io/jenkins/plugins/opentelemetry/OtelUtils.java

```diff
@@ -40,24 +40,72 @@
 
     @Nonnull
     public static String getProjectType(Run run) {
-        if (run == null) {
-            return "unknown";
-        }
-        if (run instanceof FreeStyleBuild) {
+        if (isFreestyle(run)) {
             return "freestyle";
         }
-        if (run instanceof WorkflowRun) {
-            if (run.getParent().getParent() instanceof WorkflowMultiBranchProject) {
-                return "multibranch";
-            } else {
-                return "workflow";
-            }
+        if (isMultibranch(run)) {
+            return "multibranch";
+        }
+        if (isWorkflow(run)) {
+            return "workflow";
         }
         // TODO: support for
         // https://github.com/jenkinsci/matrix-project-plugin/blob/master/src/main/java/hudson/matrix/MatrixBuild.java#L70
         return "unknown";
     }
 
+    @Nonnull
+    public static String getMultibranchType(Run run) {
+        if (isMultibranch(run)) {
+            if (isMultibranchPullRequest(run)) {
+                return "pr";
+            }
+            if (isMultibranchBranch(run)) {
+                return "branch";
+            }
+            // TODO: discover tag type.
+        }
+        return "unknown";
+    }
+
+    public static boolean isMultibranchPullRequest(Run run) {
```

Comment:
The generic concept in Jenkins is "ChangeRequest" rather than "PullRequest"

Reply:
See https://github.com/jenkinsci/opentelemetry-plugin/pull/52/commits/ecb33b2383d24d5c0e61148deeabc0e000e085d1